src/services/codefixes/inferFromUsage.ts

```diff
@@ -309,7 +308,7 @@ namespace ts.codefix {
             }
             const isConstructor = declaration.kind === SyntaxKind.Constructor;
             const callContexts = isConstructor ? usageContext.constructContexts : usageContext.callContexts;
-            return callContexts && declaration.parameters.map((parameter, parameterIndex) => {
+            return callContexts && declaration.parameters.map((parameter, parameterIndex): ParameterInference => {

```

Comment:
Return types don't get a contextual type. This made it look like `isOptional` was unused. Currently it's write-only but that should change in #27978.